### ISSUE TITLE
Use Draft<S> from immer in TUpdateFunction 

### ISSUE THREAD

**Issue by blutorange** [lostpebble/pullstate#48]
This is a great small library I just found, good work : ) There's a minor issue with TypeScript that would be great if it could be changed:

```typescript
interface S {
  readonly items: readonly string[];
}

const store = new Store<S>({items: []});

// this gives an error "splice does not exist on type ..."
store.update(draft => { draft.items.splice(1, 1) });

// with immer it works
declare const s: S;  
produce(s, draft => { draft.items.splice(1, 1) });
```

This doesn't transpile as items is a readonly array. `immer` has a `Draft` type for this case - would be great if it could be used in pullstate:

https://github.com/lostpebble/pullstate/blob/master/src/Store.ts#L26 - https://github.com/lostpebble/pullstate/blob/master/src/Store.ts#L28

```typescript
import { Draft } from "immer";

export type TUpdateFunction<S> = (draft: Draft<S>, original: S) => void;
type TPathReactionFunction<S> = (paths: TAllPathsParameter<S>, draft: Draft<S>, original: S) => void;
type TReactionFunction<S, T> = (watched: T, draft: Draft<S>, original: S, previousWatched: T) => void;

```

**Comment by lostpebble** [comment]
Hi @blutorange !

Thanks for the kind words, I'm glad you're enjoying it :)

Ah, I see. Makes sense, I'll look into putting that in now. Never made use of `readonly` before so never ran into this issue- thanks for bringing it to my attention!

**Comment by lostpebble** [comment]
I've added it and published `1.15.3` with the new type. Let me know if you run into any further issues with it.

**Comment by blutorange** [comment]
Thank you very much, that was pretty quick :) 

And yeah, I hadn't been using readonly either until recently -- but I found it's very useful to keep you (and your co-workers) from accidentally doing something bad : )

**Comment by blutorange** [comment]
Could you also update the `draft` argument for `TReactionFunction` or is that something else ? https://github.com/lostpebble/pullstate/blob/master/src/Store.ts#L28

**Comment by lostpebble** [comment]
Ah yea, that is the same. Added it and released in `1.15.4` :)
